### ISSUE TITLE
Search index does not decode &#…; with more than 4 digits

### ISSUE THREAD

**Issue by DeriAdno** [tutao/tutanota#979]
It looks like all mail bodies are passed through [IndexUtils.htmlToText](https://github.com/tutao/tutanota/blob/cee72fca7ea24c1698d4c4d37c0c49936098d29c/src/api/worker/search/IndexUtils.js#L98) to replace each `&…;` with the corresponding character before building a search index.
```js
text.replace(/&[#,0-9,a-z,A-Z]{1,5};/g, …
```
which, for numbers, allows `&#`(at most 4 digits)`;`. So, let's try to break the search with a codepoint value longer than 4 digits in base 10.
(Another potential problem is that some `HTML_ENTITIES` used by this function are longer than 5 chars and won't match the regexp, e.g. [`&Agrave;`](https://github.com/tutao/tutanota/blob/cee72fca7ea24c1698d4c4d37c0c49936098d29c/src/api/worker/search/IndexUtils.js#L117), but I haven't tested this).

- Send a plain-text email from another provider (e.g. Gmail) with this body: `⥣ Ի`.
The codepoints are: `⥣ (U+2963 &#10595;)`, `Ի (U+053B &#1339;)`.
- The email is displayed correctly.
- Typing `Ի` in the search box finds the email.
- **Typing `⥣` in the search box finds no results.**

---

I propose changing the regexp (and adding tests) to allow both:
- The largest unicode codepoint.
    [The current Unicode standard](https://www.unicode.org/versions/Unicode11.0.0/UnicodeStandard-11.0.pdf) on page 29 says:
    > In the Unicode Standard, the codespace consists of the integers from 0 to 10FFFF (base 16).

    `0x10FFFF == 1114111` so 7 digits in base 10. Plus leading `#`: 8 chars between `&` and `;`.

- The longest entry in [HTML_ENTITIES](https://github.com/tutao/tutanota/blob/cee72fca7ea24c1698d4c4d37c0c49936098d29c/src/api/worker/search/IndexUtils.js#L112), `&thetasym;`: 8 chars.

```js
text.replace(/&[#,0-9,a-z,A-Z]{1,8};/g, …
```

---

Discovered this by searching for `&#` after reporting #978 but this is a different bug in different code so I opened a separate issue.

**Comment by charlag** [comment]
Hey, thanks for reporting, this is a good find!
I'm not sure why we need this length limit at all, I would propose to change it to
`/&[#0-9a-zA-Z]+;/g`

**Comment by armhub** [comment]
Check if this will be fixed with #796
